drivers/builtin/src/entropy_poll.h

```diff
@@ -57,6 +41,21 @@ int mbedtls_nv_seed_poll(void *data,
                          unsigned char *output, size_t len, size_t *olen);
 #endif
 
+/* Wrapper to allow mbedtls_entropy_hardware_poll() to be used from entropy module
+ * as other entropy polling functions (i.e. mbedtls_platform_entropy_poll and
+ * mbedtls_nv_seed_poll). */
+#if defined(MBEDTLS_ENTROPY_HARDWARE_POLL)
+
+#include <mbedtls/platform.h>
+
+static inline int mbedtls_entropy_hardware_poll_wrapper(
+        void *data, unsigned char *output, size_t len, size_t *olen)
+{
+    (void) data;
+    return mbedtls_entropy_hardware_poll(output, len, olen, NULL);
```

Comment:
The wrapper should check the entropy content and fail if it's less than full entropy, i.e. if `entropy_output < 8 * *olen`. Please add a unit test for that in `test_suite_entropy`.

Also the entropy collection should fail if `*olen` is too small and as a result there isn't enough entropy, collected from all sources. I believe this is covered by the existing tests in `test_suite_entropy`, but please double-check.

Reply:
>The wrapper should check the entropy content and fail if it's less than full entropy, i.e. if entropy_output < 8 * *olen. Please add a unit test for that in test_suite_entropy.

While adding the check on the wrapper and testing it with the new component `tf_psa_crypto_platform_get_entropy_alt` I found that tests are already failing if the returned entropy content is not correct, so there seems to be no need for a new test step. As reference implementation I'm using the one proposed in the companion PR https://github.com/Mbed-TLS/mbedtls-framework/pull/151.